cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -633,6 +748,188 @@ typename DataHelper::NodeType::Base * TrainBatchTaskBase<algorithmFPType, DataHe
     return makeLeaf(_aSample.get() + iStart, n, curImpurity, nClasses);
 }
 
+template <typename algorithmFPType, typename DataHelper, CpuType cpu>
+typename DataHelper::NodeType::Base * TrainBatchTaskBase<algorithmFPType, DataHelper, cpu>::buildBestFirst(
```

Comment:
I insist to return the status. There are a lot of checks here. There may be security issues.

Reply:
status is an argument in buildBestFirst(..., services::Status & s,...) method